packages/react-router/docs/guides/testing.md

```diff
@@ -160,41 +108,35 @@ const App = () => (
 );
 
 // the actual test!
-it("navigates around", done => {
-  renderTestSequence({
-    // tell it the subject you're testing
-    subject: App,
-
-    // and the steps to execute each time the location changes
-    steps: [
-      // initial render
-      ({ history, div }) => {
-        // assert the screen says what we think it should
-        console.assert(div.innerHTML.match(/Welcome/));
-
-        // now we can imperatively navigate as the test
-        history.push("/dashboard");
-      },
-
-      // second render from new location
-      ({ div }) => {
-        console.assert(div.innerHTML.match(/Dashboard/));
-
-        // or we can simulate clicks on Links instead of
-        // using history.push
-        Simulate.click(div.querySelector("#click-me"), {
-          button: 0
-        });
-      },
-
-      // final render
-      ({ location }) => {
-        console.assert(location.pathname === "/");
-        // you'll want something like `done()` so your test
-        // fails if you never make it here.
-        done();
-      }
-    ]
+it("navigates home when you click the logo", async => {
+  // in a real test a renderer like "@testing-library/react"
+  // would take care of setting up the DOM elements
+  const root = document.createElement('div')
+  document.body.appendChild(root);
+  
+  // Set initial location
+  window.history.pushState({}, '', '/my/initial/route');
+  
+  // Render app
+  render(
+    <BrowserRouter>
+      <App />
+    <BrowserRouter>,
+    root
+  )
+  
+  // Interact with page
+  act(() => {
+    // Find the link (perhaps using the text content)
+    const goHomeLink = document.querySelector('#nav-logo-home');
+    // Click it
+    goHomeLink.dispatchEvent(new MouseEvent("click", { bubbles: true }));
```

Comment:
Wouldn't it be easier to `Simulate.click(goHomeLink)` here?

Reply:
Kent is right about it here https://github.com/ReactTraining/react-router/pull/6903#issuecomment-530141492